File: test_login.js
// Test login script to verify authentication
const axios = require('axios');
const fs = require('fs');

const API_URL = 'http://localhost:8000/api/v1';
const email = 'ana.petrescu@student.usv.ro';
const password = 'Student2025!';

async function testLogin() {
  try {
    console.log(`Attempting to login with ${email}`);
    
    const formData = new URLSearchParams();
    formData.append('username', email);
    formData.append('password', password);
    
    const response = await axios.post(`${API_URL}/auth/login`, formData, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
      timeout: 10000
    });
    
    if (response.data && response.data.access_token) {
      console.log('Login successful!');
      console.log('Token:', response.data.access_token);
      
      // Save token to file for testing
      fs.writeFileSync('token.txt', response.data.access_token);
      
      // Test token by making a request to get user info
      try {
        const userResponse = await axios.get(`${API_URL}/users/me`, {
          headers: {
            'Authorization': `Bearer ${response.data.access_token}`
          }
        });
        
        console.log('User info retrieved successfully:');
        console.log(JSON.stringify(userResponse.data, null, 2));
      } catch (userError) {
        console.error('Error fetching user info:', userError.message);
        if (userError.response) {
          console.error('Response status:', userError.response.status);
          console.error('Response data:', userError.response.data);
        }
      }
    } else {
      console.error('No token in response:', response.data);
    }
  } catch (error) {
    console.error('Login error:', error.message);
    if (error.response) {
      console.error('Response status:', error.response.status);
      console.error('Response data:', error.response.data);
    }
  }
}

testLogin();

File: test_login.py
import requests
import json

# API endpoint
login_url = "http://localhost:8000/api/v1/auth/login"

# Test credentials
credentials = [
    {"username": "secretariat@usv.ro", "password": "password123"},
    {"username": "secretariat@usv.ro", "password": "secretariat123"},
    {"username": "admin@example.com", "password": "password123"},
    {"username": "professor@example.com", "password": "password123"}
]

# Test each set of credentials
for cred in credentials:
    print(f"\nTesting login with: {cred['username']}")
    
    # Format data as form data
    form_data = {
        "username": cred["username"],
        "password": cred["password"]
    }
    
    try:
        # Send POST request
        response = requests.post(login_url, data=form_data)
        
        # Print response details
        print(f"Status Code: {response.status_code}")
        print(f"Response: {response.text}")
        
        if response.status_code == 200:
            print("Login successful!")
            token = response.json().get("access_token")
            print(f"Token: {token[:20]}...")
            
            # Test getting user info with token
            user_url = "http://localhost:8000/api/v1/users/me"
            headers = {"Authorization": f"Bearer {token}"}
            
            user_response = requests.get(user_url, headers=headers)
            print(f"\nUser Info Status: {user_response.status_code}")
            if user_response.status_code == 200:
                print(f"User Info: {user_response.json()}")
            else:
                print(f"Failed to get user info: {user_response.text}")
        else:
            print("Login failed!")
    
    except Exception as e:
        print(f"Error: {str(e)}")

print("\nLogin testing completed.")
